Add new counters that were added in .NET 5 (#20821)
dotnet/docs: docs/core/diagnostics/includes/available-counters.md

| :::no-loc text="Gen 2 GC Count"::: (`gen-2-gc-count`) | The number of times GC has occurred for Gen 2 |
| :::no-loc text="Gen 2 Size"::: (`gen-2-size`) | The number of bytes for Gen 2 GC |
| :::no-loc text="LOH Size"::: (`loh-size`) | The number of bytes for Gen 3 GC |

> | :::no-loc text="POH Size"::: (`poh-size`) | The number of bytes for Pinned Object Heap (Available on .NET 5 or later) |

Reviewer: ```suggestion
| :::no-loc text="POH Size"::: (`poh-size`) | The number of bytes for the pinned object heap (available on .NET 5 and later versions) |
```

> | :::no-loc text="GC Fragmentation"::: (`gc-fragmentation`) | The GC Heap Fragmentation (Available on .NET 5 or later) |

Reviewer: ```suggestion
| :::no-loc text="GC Fragmentation"::: (`gc-fragmentation`) | The GC heap fragmentation (available on .NET 5 and later versions) |
```

| :::no-loc text="Number of Assemblies Loaded"::: (`assembly-count`) | The number of <xref:System.Reflection.Assembly> instances loaded into a process at a point in time |
| :::no-loc text="ThreadPool Completed Work Item Count"::: (`threadpool-completed-items-count`) | The number of work items that have been processed so far in the <xref:System.Threading.ThreadPool> |
| :::no-loc text="ThreadPool Queue Length"::: (`threadpool-queue-length`) | The number of work items that are currently queued to be processed in the <xref:System.Threading.ThreadPool> |
| :::no-loc text="ThreadPool Thread Count"::: (`threadpool-thread-count`) | The number of thread pool threads that currently exist in the <xref:System.Threading.ThreadPool>, based on <xref:System.Threading.ThreadPool.ThreadCount?displayProperty=nameWithType> |
| :::no-loc text="Working Set"::: (`working-set`) | The amount of physical memory mapped to the process context at a point in time base on <xref:System.Environment.WorkingSet?displayProperty=nameWithType> |

> | :::no-loc text="IL Bytes Jitted"::: (`il-bytes-jitted`) | The total size of IL that are JIT-compiled in bytes. (Available on .NET 5 or later) |

Reviewer: ```suggestion
| :::no-loc text="IL Bytes Jitted"::: (`il-bytes-jitted`) | The total size of IL that's JIT-compiled, in bytes (available on .NET 5 and later versions) |
```

> | :::no-loc text="Method Jitted Count"::: (`method-jitted-count`) | The number of methods that are JIT-compiled. (Available on .NET 5 or later) |

Reviewer: ```suggestion
| :::no-loc text="Method Jitted Count"::: (`method-jitted-count`) | The number of methods that are JIT-compiled (available on .NET 5 and later versions) |
```
